File: javascript/rna-transcription/rna-transcription.js
const RnaTranscriber = function() {
};

function convert(dnaNucleotice) {
  switch (dnaNucleotice) {
    case 'G': return 'C';
    case 'C': return 'G';
    case 'T': return 'A';
    case 'A': return 'U';
    default: throw new Error('Invalid input');
  }
}

RnaTranscriber.prototype.toRna = function(input) {
  return input.split('').map(convert).join('');
};

module.exports = RnaTranscriber;

File: python/rna-transcription/rna_transcription.py
def convert(nucleotide):
    if nucleotide == 'G':
        return 'C'
    if nucleotide == 'C':
        return 'G'
    if nucleotide == 'T':
        return 'A'
    if nucleotide == 'A': return 'U'


def to_rna(dna_strand):
    return ''.join(map(convert, dna_strand))


